src/jobflow/core/job.py

```diff
@@ -927,6 +926,7 @@ def update_metadata(
         function_filter: Callable = None,
         dict_mod: bool = False,
         dynamic: bool = True,
+        callback_filter: Callable[[jobflow.Flow | Job], bool] = lambda _: True,
```

Comment:
@janosh: the `Callable` `callback_filter`s seem to break some functionality, getting `'dict' object is not callable` when these are decoded. Double checked manually that `MontyDecoder` is not rehydrating these into callables, just as `dict`s

Reply:
Or you issue an error when someone tries to serialize a callback_filter to explain that this feature is only compatible with create-and-run kind of workflows

Reply:
Serialisation shouldn't be a problem if the function is importable, e.g if it is defined in atomate2 or another package.

Reply:
The default `callback_filter` isn't serializing correctly - I'll look into it a bit more and update you both